spec/lib/traject/config/resource_type_spec.rb

```diff
@@ -0,0 +1,266 @@
+        let(:record) do
+          MARC::Record.new.tap do |r|
+            r.leader = '000000g00000000000000000'
+            r.append(MARC::ControlField.new('008', '000000000000000000000000000000000k0000'))
+          end
+        end
+
+        it 'maps to Image' do
+          expect(result['resource_type_ssim']).to eq ['Image']
+        end
+      end
+
+      context 'when record is an Image based on 245h terms' do
+        let(:record) do
+          MARC::Record.new.tap do |r|
+            r.leader = '000000a00000000000000000'
+            r.append(MARC::DataField.new('245', '1', ' ',
+                                         MARC::Subfield.new('a', 'Example title'),
+                                         MARC::Subfield.new('h', 'technical drawing')))
+          end
+        end
+
+        it 'maps to Image' do
+          expect(result['resource_type_ssim']).to eq ['Image']
+        end
+      end
+
+      context 'when record is an Image based on 007 and 245h = kit' do
+        let(:record) do
+          MARC::Record.new.tap do |r|
+            r.leader = '000000a00000000000000000'
+            r.append(MARC::ControlField.new('007', 'k0000000000'))
+            r.append(MARC::DataField.new('245', '1', ' ',
+                                         MARC::Subfield.new('a', 'Example kit title'),
+                                         MARC::Subfield.new('h', 'kit')))
+          end
+        end
+
+        it 'maps to Image' do
+          expect(result['resource_type_ssim']).to eq ['Image']
+        end
+      end
+
+      context 'when record is an Image|Photo' do
+        context 'when 007[0] = k and 007[1] in [g, h, r, v]' do
+          let(:record) do
+            MARC::Record.new.tap do |r|
+              r.leader = '000000a00000000000000000'
+              r.append(MARC::ControlField.new('007', 'kg0000000000'))
+            end
+          end
+
+          it 'maps to Image|Photo' do
+            expect(result['resource_type_ssim']).to eq ['Image|Photo']
+          end
+        end
+      end
+
+      context 'when record is an Image|Poster' do
+        context 'when 007[0] = k and 007[1] = k' do
+          let(:record) do
+            MARC::Record.new.tap do |r|
+              r.leader = '000000a00000000000000000'
+              r.append(MARC::ControlField.new('007', 'kk0000000000'))
+            end
+          end
+
+          it 'maps to Image|Poster' do
+            expect(result['resource_type_ssim']).to eq ['Image|Poster']
+          end
+        end
+      end
+
+      context 'when record is an Image|Slide' do
+        context 'when 007[0] = g and 007[1] = s' do
+          let(:record) do
+            MARC::Record.new.tap do |r|
+              r.leader = '000000a00000000000000000'
+              r.append(MARC::ControlField.new('007', 'gs0000000000'))
+            end
+          end
+
+          it 'maps to Image|Slide' do
+            expect(result['resource_type_ssim']).to eq ['Image|Slide']
```

Comment:
For the hierarchy to work, we need both terms, e.g.:

```suggestion
            expect(result['resource_type_ssim']).to eq ['Image', 'Image|Slide']
```